exchanges/graphcache/src/cacheExchange.ts

```diff
@@ -96,39 +96,55 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
     });
   }
 
-  const optimisticKeysToDependencies = new Map<number, Set<string>>();
+  const optimisticKeysToDependencies: OptimisticDependencies = new Map();
+  const mutationResultBuffer: OperationResult[] = [];
   const ops: OperationMap = new Map();
+  const blockedDependencies: Dependencies = makeDict();
+  const requestedRefetch: Operations = new Set();
   const deps: DependentOperations = makeDict();
 
+  const isBlockedByOptimisticUpdate = (dependencies: Dependencies) => {
+    for (const dep in dependencies) {
+      if (blockedDependencies[dep]) {
+        return true;
+      }
```

Comment:
```suggestion
      return !!blockedDependencies[dep])
```

Reply:
Oh actually, we have to keep iterating here if it's not in there to check all dependencies 😅